internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -160,63 +160,72 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context, clust
-	updateString(spec.BaseDomain, c.BaseDNSDomain, &params.BaseDNSDomain)
+	updateString(spec.ClusterName, c.Name, &params.Name, false)
+	updateString(spec.BaseDomain, c.BaseDNSDomain, &params.BaseDNSDomain, false)
 
 	if len(spec.Provisioning.InstallStrategy.Agent.Networking.ClusterNetwork) > 0 {
-		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.ClusterNetwork[0].CIDR, c.ClusterNetworkCidr, &params.ClusterNetworkCidr)
+		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.ClusterNetwork[0].CIDR, c.ClusterNetworkCidr, &params.ClusterNetworkCidr, false)
 		hostPrefix := int64(spec.Provisioning.InstallStrategy.Agent.Networking.ClusterNetwork[0].HostPrefix)
 		if hostPrefix > 0 && hostPrefix != c.ClusterNetworkHostPrefix {
 			params.ClusterNetworkHostPrefix = swag.Int64(hostPrefix)
 			update = true
 		}
 	}
 	if len(spec.Provisioning.InstallStrategy.Agent.Networking.ServiceNetwork) > 0 {
-		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.ServiceNetwork[0], c.ServiceNetworkCidr, &params.ServiceNetworkCidr)
+		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.ServiceNetwork[0], c.ServiceNetworkCidr, &params.ServiceNetworkCidr, false)
 	}
 	if len(spec.Provisioning.InstallStrategy.Agent.Networking.MachineNetwork) > 0 {
-		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.MachineNetwork[0].CIDR, c.MachineNetworkCidr, &params.MachineNetworkCidr)
+		updateString(spec.Provisioning.InstallStrategy.Agent.Networking.MachineNetwork[0].CIDR, c.MachineNetworkCidr, &params.MachineNetworkCidr, false)
 	}
 
-	updateString(spec.Platform.AgentBareMetal.APIVIP, c.APIVip, &params.APIVip)
-	updateString(spec.Platform.AgentBareMetal.APIVIPDNSName, swag.StringValue(c.APIVipDNSName), &params.APIVipDNSName)
-	updateString(spec.Platform.AgentBareMetal.IngressVIP, c.IngressVip, &params.IngressVip)
-	updateString(spec.Provisioning.InstallStrategy.Agent.SSHPublicKey, c.SSHPublicKey, &params.SSHPublicKey)
+	updateString(spec.Platform.AgentBareMetal.APIVIP, c.APIVip, &params.APIVip, false)
+	updateString(spec.Platform.AgentBareMetal.APIVIPDNSName, swag.StringValue(c.APIVipDNSName), &params.APIVipDNSName, false)
+	updateString(spec.Platform.AgentBareMetal.IngressVIP, c.IngressVip, &params.IngressVip, false)
+	updateString(spec.Provisioning.InstallStrategy.Agent.SSHPublicKey, c.SSHPublicKey, &params.SSHPublicKey, false)
+
+	installEnv, err := getInstallEnvByClusterDeployment(ctx, r.Client, cluster)
 
-	// TODO: get from AgentEnvSpec
-	//updateString(spec.HTTPProxy, c.HTTPProxy, &params.HTTPProxy)
-	//updateString(spec.HTTPSProxy, c.HTTPSProxy, &params.HTTPSProxy)
-	//updateString(spec.NoProxy, c.NoProxy, &params.NoProxy)
-	//updateString(spec.AdditionalNtpSource, c.AdditionalNtpSource, &params.AdditionalNtpSource)
+	// It is possible that the clusterDeployment doesn't have an installEnv. ClusterDeploymentsReconciler should not fail for that reason.
+	if err != nil && !gorm.IsRecordNotFoundError(err) {
```

Comment:
You are not getting `InstallEnv` from postgres the error can't be of this type. `!gorm.IsRecordNotFoundError(err)`

Reply:
Done. used `k8serrors.IsNotFound(err)` instead